ceph-ansible-pr-syntax-check/build/build

```diff
@@ -82,6 +82,37 @@ function test_ceph_release_in_ceph_default {
   echo "No '- ceph_release_num[ceph_release]' statements found in ceph-defaults role!" && return 0
 }
 
+function check_if_a_keyword_is_used() {
+  searchstring=$1
+
+  # Since we are looking for keywords, instances of $searchstring found
+  # in comments and strings are not qualified. Ansible keywords are only
+  # be preceded by bunch of spaces or with a hypen somewhere between spaces.
+  # So, let's look for only those instances.
+  #
+  # Besides, completely ignore the lines marked as deleted in diff.
+  searchstring_regex="^\+ *-? +$searchstring"
+  count=$(git diff master..HEAD | grep -cE "$searchstring_regex")
+  if [[ $count > 0 ]]; then
+    echo "\"$searchstring\" is amongst the forbidden keywords; found $count times"
+    return $count
+  fi
+
+  return 0
+}
+
+function check_for_forbidden_keywords() {
```

Comment:
Is this function necessary? Why don't we simply call `check_if_a_keyword_is_used` out of a function here https://github.com/ceph/ceph-build/pull/1120/files#diff-6cbf1ad8bb8686f24d82a5d4412b4311R127? If you find something forbidden in `check_if_a_keyword_is_used` then simply return `1` instead of `count` otherwise just keep it 0.

What do you think?

Reply:
I've written two functions because I think `check_if_a_keyword_is_used` might come into use separately (since it's purpose is more generic). However, if we don't want that I can merge them into one. Should I do so?